### ISSUE TITLE
Fix seeds while loss still has fluctuation

### ISSUE THREAD

**Issue by Limingxing00** [open-mmlab/mmdetection#7351]
Hi,

I have fixed seeds by `--seed 777 --deterministic` with the **same network and same config**.   

However, I found the loss function still has fluctuation.  Specifically, they have the same trend but there are slight differences in value (~1%). 

I'm confused why the loss function still has tiny fluctuations.

**Comment by jbwang1997** [comment]
Even deterministic=True is set, the backpropagation may also be non-deterministic due to some CUDA implementation, but the difference is small.

**Comment by Limingxing00** [comment]
Thank you! I get it.
